applications/GeoMechanicsApplication/custom_utilities/generic_utilities.h

```diff
@@ -48,6 +49,36 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GenericUtilities
         return result;
     }
 
+    template <class ContainerType, class IdsType>
+    static void CollectIdsFromEntity(const ContainerType& rContainer, IdsType& rIds)
+    {
+        for (const auto& r_item : rContainer) {
+            const auto id = r_item.Id();
+
+            if constexpr (requires { rIds.insert(id); }) {
+                // Works for set/unordered_set
+                rIds.insert(id);
+            } else if constexpr (requires { rIds.push_back(id); }) {
+                // Works for vector/deque
+                rIds.push_back(id);
+            } else {
+                static_assert(!std::is_same_v<IdsType, IdsType>,
+                              "OutputContainerType does not support insert or push_back.");
+            }
+        }
+    }
+
+    template <class ContainerType>
+    static auto CollectIdsFromEntity(const ContainerType& rContainer)
```

Comment:
Perhaps we can change the name a bit to reflect that we get the IDs of the entities inside the container, e.g.:
```suggestion
    static auto GetIdsOfEntitiesIn(const ContainerType& rContainer)
```
If you agree with this suggestion, then please also rename the other overload.

Reply:
GetIds is definitely better. What do you think about `GetIdsFromEntityContents`?

Reply:
Yes, the name you've proposed reflects better that we are looking at the items inside the container. Thanks for changing this function name.